qiskit_nature/operators/second_quantization/quadratic_hamiltonian.py

```diff
@@ -0,0 +1,306 @@
+
+        Returns:
+            - The matrix :math:`A`
+            - The constant
+        """
+        original = np.block(
+            [
+                [self.antisymmetric_part, self.hermitian_part],
+                [-self.hermitian_part.conj(), -self.antisymmetric_part.conj()],
+            ]
+        )
+        eye = np.eye(self._n_orbitals, dtype=complex)
+        majorana_basis = np.block([[eye, eye], [1j * eye, -1j * eye]]) / np.sqrt(2)
+        matrix = -1j * majorana_basis.conj() @ original @ majorana_basis.T.conj()
+        constant = 0.5 * np.trace(self.hermitian_part) + self.constant
+        # imaginary parts should be zero
+        return np.real(matrix), np.real(constant)
+
+    def diagonalizing_bogoliubov_transform(self) -> Tuple[np.ndarray, np.ndarray, float]:
+        r"""Return the transformation matrix that diagonalizes a quadratic Hamiltonian.
+
+        Recall that a quadratic Hamiltonian has the form
+
+        .. math::
+            \sum_{p, q} M_{pq} a^\dagger_p a_q
+            + \frac12 \sum_{p, q}
+                (\Delta_{pq} a^\dagger_p a^\dagger_q + \text{h.c.})
+            + \text{constant}
+
+        where the :math:`a^\dagger_j` are fermionic creation operations.
+        A quadratic Hamiltonian can always be rewritten in the form
+
+        .. math::
+            \sum_{j} \varepsilon_j b^\dagger_j b_j + \text{constant},
+
+        where the :math:`b^\dagger_j` are a new set of fermionic creation operators
+        that also satisfy the canonical anticommutation relations.
+        These new creation operators are linear combinations of the old ladder
+        operators. When the Hamiltonian conserves particle number (:math:`\Delta = 0`)
+        then only creation operators need to be mixed together:
+
+        .. math::
+           \begin{pmatrix}
+                b^\dagger_1 \\
+                \vdots \\
+                b^\dagger_N \\
+           \end{pmatrix}
+           = W
+           \begin{pmatrix}
+                a^\dagger_1 \\
+                \vdots \\
+                a^\dagger_N \\
+           \end{pmatrix},
+
+        where :math:`W` is an :math:`N \times N` unitary matrix. However, in the general case,
+        both creation and annihilation operators are mixed together:
+
+        .. math::
+           \begin{pmatrix}
+                b^\dagger_1 \\
+                \vdots \\
+                b^\dagger_N \\
+           \end{pmatrix}
+           = W
+           \begin{pmatrix}
+                a^\dagger_1 \\
+                \vdots \\
+                a^\dagger_N \\
+                a_1 \\
+                \vdots \\
+                a_N
+           \end{pmatrix},
+
+        where now :math:`W` is an :math:`N \times 2N` matrix with orthonormal rows
+        (which satisfies additional constraints).
+
+        Returns:
+            - The matrix :math:`W`, which is either an :math:`N \times N` or
+              an :math:`N \times 2N` matrix
```

Comment:
Maybe repeat that the latter is the case the `\Delta != 0`

Reply:
Done.